File: CS/T449201/Controllers/HomeController.cs
using DevExpress.Web.Mvc;
using DevExpress.Web.Office;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Spreadsheet;


namespace T449201.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult SpreadsheetPartial() {
            return PartialView("_SpreadsheetPartial");
        }
        public ActionResult SpreadsheetCustomPartial(string diagramName) {
            IWorkbook wbook =  SpreadsheetExtension.GetCurrentDocument("Spreadsheet");
            
            switch (diagramName) {
                case "PieChart":
                    DiagramCreationHelper.CreatePieChart(wbook);
                    break;
                case "BarChart":
                    DiagramCreationHelper.CreateBarChart(wbook);
                    break;
                case "ColumnChart":
                    DiagramCreationHelper.CreateColumnChart(wbook);
                    break;
                case "ComplexChart":
                    DiagramCreationHelper.CreateComplexChart(wbook);
                    break;
                case "DoughnutChart":
                    DiagramCreationHelper.CreateDoughnutChart(wbook);
                    break;
                case "Pie3dChart":
                    DiagramCreationHelper.CreatePie3dChart(wbook);
                    break;
                case "ScatterChart":
                    DiagramCreationHelper.CreateScatterChart(wbook);
                    break;
                case "StockChart":
                    DiagramCreationHelper.CreateStockChart(wbook);
                    break;
                case "BubbleChart":
                    DiagramCreationHelper.CreateBubbleChart(wbook);
                    break;
            }

                    return PartialView("_SpreadsheetPartial");
        }
    }
}

File: VB/T449201/Controllers/HomeController.vb
Imports DevExpress.Web.Mvc
Imports DevExpress.Web.Office
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Spreadsheet


Namespace T449201.Controllers
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function SpreadsheetPartial() As ActionResult
			Return PartialView("_SpreadsheetPartial")
		End Function
		Public Function SpreadsheetCustomPartial(ByVal diagramName As String) As ActionResult
			Dim wbook As IWorkbook = SpreadsheetExtension.GetCurrentDocument("Spreadsheet")

			Select Case diagramName
				Case "PieChart"
					DiagramCreationHelper.CreatePieChart(wbook)
				Case "BarChart"
					DiagramCreationHelper.CreateBarChart(wbook)
				Case "ColumnChart"
					DiagramCreationHelper.CreateColumnChart(wbook)
				Case "ComplexChart"
					DiagramCreationHelper.CreateComplexChart(wbook)
				Case "DoughnutChart"
					DiagramCreationHelper.CreateDoughnutChart(wbook)
				Case "Pie3dChart"
					DiagramCreationHelper.CreatePie3dChart(wbook)
				Case "ScatterChart"
					DiagramCreationHelper.CreateScatterChart(wbook)
				Case "StockChart"
					DiagramCreationHelper.CreateStockChart(wbook)
				Case "BubbleChart"
					DiagramCreationHelper.CreateBubbleChart(wbook)
			End Select

					Return PartialView("_SpreadsheetPartial")
		End Function
	End Class
End Namespace